t/integration/test_canvas.py

```diff
@@ -969,7 +969,7 @@ def test_callback_called_by_group(self, manager, subtests):
             raise pytest.skip("Requires redis result backend.")
         redis_connection = get_redis_connection()
 
-        callback_msg = str(uuid.uuid4()).encode()
```

Comment:
@thedrow this needs to be changed for the message.body serialization

Reply:
How exactly and why?

Reply:
Signature args in callback_msg should be serializable, so
```
callback_msg = str(uuid.uuid4()).encode()
redis_key = str(uuid.uuid4())
callback = redis_echo.si(callback_msg, redis_key=redis_key)
```
,so should be
```
callback_msg = str(uuid.uuid4())
redis_key = str(uuid.uuid4())
callback = redis_echo.si(callback_msg, redis_key=redis_key)

```
as they will be passed inside as_task_v2 that would return message that would be passed to a send_task_message, then to publish (kombu) and _prepare (kombu) in messaging that would cll a serialization (kombu).
```
from kombu.serialization import dumps
s = ((42,), {}, {'callbacks': [{'task': 't.integration.tasks.redis_echo', 'args': (b'7a108737-0b0a-4acc-a32f-f8a5e6726775',), 'kwargs': {'redis_key': 'ef3bc79f-a04c-4de1-88c0-f13d9d999ca8'}, 'options': {}, 'subtask_type': None, 'immutable': True}], 'errbacks': None, 'chain': None, 'chord': None})
print(dumps(s))
```
to reproduce serialization issue